chart/templates/post-delete-hook-cluster-role.yaml

```diff
@@ -30,11 +30,11 @@ rules:
   - apiGroups: [ "admissionregistration.k8s.io" ]
     resources: [ "validatingwebhookconfigurations", "mutatingwebhookconfigurations" ]
     verbs: [ "get", "list", "delete" ]
-  {{- if .Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy" }}
+{{- if .Values.global.cattle.psp.enabled }}
   - apiGroups: [ "policy" ]
     resources: [ "podsecuritypolicies" ]
-    verbs: [ "delete", "create" ]
-  {{- end }}
+    verbs: [ "use", "delete", "create" ]
```

Comment:
I don't think that this approach quite works. Even if this chart doesn't have PSPs enabled, I could see the post-delete job needing to delete PSPs for other charts which are using PSPs (like a feature chart in the local cluster). You may want to consider separating these into two rules - one for the delete/create permission and one for the use permission, and then only use the if check around the use permission.

Reply:
Makes sense. Done.